### PR TITLE
Dont auto scroll upon mouseclick

### PR DESCRIPTION
**Pull request by RensDofferhoff**
Focus event chains are a little weird so had to be more specific

### CODE REVIEW

File: Desktop/components/JASP/Widgets/AnalysisFormExpander.qml [hunk 1]
```diff
@@ -225,7 +225,7 @@ DropArea
 
 					backgroundFlickable.scrollToElement(expanderButton);
 					if(firstExpansion) //only focus first item on analysis creation
-						formParent.nextItemInFocusChain().forceActiveFocus();
+						formParent.nextItemInFocusChain().forceActiveFocus(Qt.OtherFocusReason);
```
Comment (boutinb) [inline Desktop/components/JASP/Widgets/AnalysisFormExpander.qml]: Why this change, since OtherFocusReason is the default?

File: Desktop/components/JASP/Widgets/AnalysisFormExpander.qml [hunk 2]
```diff
@@ -595,7 +595,8 @@ DropArea
 								return;
 
 							const control = myForm.activeJASPControl;
-							if (control.focusReason !== Qt.MouseFocusReason)
+							var x =  control.focusReason;
+							if (control.focusReason === Qt.TabFocusReason || control.focusReason === Qt.BacktabFocusReason || control.focusReason === Qt.OtherFocusReason)
```
Comment (boutinb) [inline Desktop/components/JASP/Widgets/AnalysisFormExpander.qml]: Why this change again? Which focus reasons you don't want here? With this change you exclude now MouseFocusReason, but also ActiveWindowFocusReason, PopupFocusReason, ShortcutFocusReason and MenuBarFocusReason. Is the ShortcutFocusReason not a good reason for scrolling?

**Review by boutinb — CHANGES_REQUESTED** [review]

**Comment by RensDofferhoff** [comment]
Fixed the fix @boutinb.
Good suggestions

File: QMLComponents/controls/jaspcontrol.cpp [hunk 3]
```diff
@@ -793,6 +793,9 @@ void JASPControl::_notifyFormOfActiveFocus()
 	if (!hasActiveFocus())
 		_hasActiveFocus = false;
 
-	if (_form)
+	//All JASP controls are focusscopes when they receive focus due to a child being clicked the reason is 8 for some reason (undocumented as of 3-1-2023)
+	//Objects like jasp groupboxes could unrightfully be marked as the active jasp control when a child jasp control gets focus
+	//For this reason we check if the focus reason is the result of user input or the focus scope system.
+	if (_form && _focusReason >= Qt::MouseFocusReason && _focusReason <= Qt::OtherFocusReason)
```
Comment (boutinb) [inline QMLComponents/controls/jaspcontrol.cpp:799]: Sorry, I don't understand this extra check : the only focus reason that cannot be between MouseFocusReason and OtherFocusReason is NoFocusReason (also, as this function is called only when the focus changes, that would be really strange).

**Review by boutinb — CHANGES_REQUESTED** [review]

File: QMLComponents/controls/jaspcontrol.cpp (hunk 3 above)
Comment (RensDofferhoff) [reply QMLComponents/controls/jaspcontrol.cpp:799]: There are hidden values for the FocusReason Enum that do not concern user input.
Such as when a FocusScope gets focus due to a focus event of  one of its children (value 8) .
You can check it under the debugger. It is weird but true.

**Review by boutinb — APPROVED** [review]
